engine/src/execution_graph/logic_controllers/taskflow/graph.h

```diff
@@ -0,0 +1,206 @@
+#pragma once
+
+#include "kernel.h"
+#include "kpair.h"
+
+namespace ral {
+namespace cache {
+
+
+static std::shared_ptr<ral::cache::CacheMachine> create_cache_machine(const cache_settings& config) {
+	std::shared_ptr<ral::cache::CacheMachine> machine;
+	if (config.type == CacheType::NON_WAITING) {
+		machine =  std::make_shared<ral::cache::NonWaitingCacheMachine>();
+	}
+	else if (config.type == CacheType::SIMPLE or config.type == CacheType::FOR_EACH) {
+		machine =  std::make_shared<ral::cache::CacheMachine>();
+	} else if (config.type == CacheType::CONCATENATING) {
+		machine =  std::make_shared<ral::cache::ConcatenatingCacheMachine>();
+	}
+	return machine;
+}
+
+static std::vector<std::shared_ptr<ral::cache::CacheMachine>> create_cache_machines(const cache_settings& config) {
```

Comment:
I am loathe to the cache machiens being put into a vector likek this. In all honesty I would much prefer a graph where the cache maciens and transform engines are actually connected to each other and both implement a "NODE" interface or somethign like that.

Reply:
That being said this is just a comment. mothing to do right now